### PR TITLE
[BLOCKED] Add CLIP operation

### PR DESCRIPTION
**Pull request by Benny-Nottonson**
This pull request adds a new CLIP operation to the codebase. The CLIP operation allows for clipping the values of a tensor within a specified range. This can be useful for tasks such as gradient clipping in neural networks.

### CODE REVIEW

**Comment by soraros** [comment]
Can we implement it as `min(max(x, ub), lb)`?

**Comment by Benny-Nottonson** [comment]
> Can we implement it as `min(max(x, ub), lb)`?

Wouldnt that be the same as `x.max(ub).min(lb)` ?

**Comment by soraros** [comment]
Yea, my point is, do we need a new Op?

**Comment by Benny-Nottonson** [comment]
> Yea, my point is, do we need a new Op?

It may be a little bit of syntactic sugar but I think its cleaner to give it as an option rather than forcing max and min seperately

**Comment by soraros** [comment]
I'm definitely not against providing wrapper functions with clear semantics. I think we should have a concept of "compositional op", so we reuse the impl and AD.

**Comment by Benny-Nottonson** [comment]
> I'm definitely not against providing wrapper functions with clear semantics. I think we should have a concept of "compositional op", so we reuse the impl and AD.

I don't see the inherent benefit in implementing this specific function as a composition of Min and Max since it is fairly simple, but im open to everyone elses thoughts on this.

**Comment by andresnowak** [comment]
Sorry for my ignorance but when saying compositional op, what we want to say is that CLIP should be ```x.max(ub).min(lb)```, instead of making it a primitive operator?.

**Comment by andresnowak** [comment]
Okay. Then for me the only thing missing is that I would say this operator should be in mlops instead of basics, And to change the test from test_ops to test_mlops, and to also have a test_mlops_torch.
